plugins/woocommerce/includes/class-wc-cart-session.php

```diff
@@ -95,33 +94,33 @@ public function init() {
 	 * @since 3.2.0
 	 */
 	public function get_cart_from_session() {
+		/**
+		 * Fires when cart is loaded from session.
+		 *
+		 * @since 3.2.0
+		 */
 		do_action( 'woocommerce_load_cart_from_session' );
-		$this->cart->set_totals( WC()->session->get( 'cart_totals', null ) );
+
+		$cart        = (array) array_filter( WC()->session->get( 'cart', array() ) );
+		$cart_totals = WC()->session->get( 'cart_totals', null );
+
+		$this->cart->set_totals( $cart_totals );
 		$this->cart->set_applied_coupons( WC()->session->get( 'applied_coupons', array() ) );
 		$this->cart->set_coupon_discount_totals( WC()->session->get( 'coupon_discount_totals', array() ) );
 		$this->cart->set_coupon_discount_tax_totals( WC()->session->get( 'coupon_discount_tax_totals', array() ) );
 		$this->cart->set_removed_cart_contents( WC()->session->get( 'removed_cart_contents', array() ) );
 
-		$update_cart_session = false; // Flag to indicate the stored cart should be updated.
-		$order_again         = false; // Flag to indicate whether this is a re-order.
-		$cart                = WC()->session->get( 'cart', null );
-		$merge_saved_cart    = (bool) get_user_meta( get_current_user_id(), '_woocommerce_load_saved_cart_after_login', true );
+		// Flag to indicate the stored cart should be updated. If cart totals are null, this will be true to calculate totals.
+		$update_cart_session = is_null( $cart_totals );
 
-		// Merge saved cart with current cart.
-		if ( is_null( $cart ) || $merge_saved_cart ) {
-			$saved_cart          = $this->get_saved_cart();
-			$cart                = is_null( $cart ) ? array() : $cart;
-			$cart                = array_merge( $saved_cart, $cart );
-			$update_cart_session = true;
-
-			delete_user_meta( get_current_user_id(), '_woocommerce_load_saved_cart_after_login' );
-		}
```

Comment:
Why was this deleted?

Reply:
So this was moved to migrate?

Reply:
We're not using the user meta data. So merging has moved to when the session is initialized.